pkg/metrics/series.go

```diff
@@ -388,8 +394,18 @@ func marshalSplitCompress(iterator serieIterator, bufferContext *marshaler.Buffe
 			return nil, err
 		}
 
-		// Compress the protobuf metadata and the marshaled series
-		err = compressor.AddItem(buf.Bytes())
+		if pointsThisPayload > maxPointsPerPayload {
```

Comment:
Consider the following code 

```go
func TestMarshalSplitCompressPointsLimitOne(t *testing.T) {
	mockConfig := config.Mock()
	oldMax := mockConfig.GetInt("serializer_max_series_points_per_payload")
	defer mockConfig.Set("serializer_max_series_points_per_payload", oldMax)
	mockConfig.Set("serializer_max_series_points_per_payload", 1)

	series := makeSeries(1, 2)
	payloads, err := series.MarshalSplitCompress(marshaler.DefaultBufferContext())
	require.NoError(t, err)
	require.Len(t, payloads, 0)
}
```
A serie with 2 points is created. As `serializer_max_series_points_per_payload` is `1`, the serie should be dropped (`stream.ErrItemTooBig`) whereas `MarshalSplitCompress` returns 2 payloads. Is it expected?

Reply:
Good point, and thanks for the test case!  I've copied the test case and fixed the underlying issue (and two more).